client/components/mma/identity/idapi/avatar.ts

```diff
@@ -46,24 +46,35 @@ const toAvatarError = (e: AvatarAPIErrorResponse): AvatarError => {
 	} as AvatarError;
 };
 
-const avatarFetch = APIFetch(IdentityLocations.AVATAR);
-
 export const read = async () => {
-	const url = '/v1/avatars/user/me/active';
-	const options = APIUseCredentials({});
-	try {
-		return await avatarFetch(url, options);
-	} catch (e) {
-		throw isAvatarAPIErrorResponse(e) ? toAvatarError(e) : e;
+	const url = '/aapi/avatar';
+	const response = await fetchWithDefaultParameters(url).then((res) =>
+		res.json(),
+	);
+	if (isAvatarAPIErrorResponse(response)) {
+		throw toAvatarError(response);
 	}
+	return response;
 };
 
 export const write = async (file: File) => {
-	const url = '/v1/avatars';
-	const options = APIUseCredentials(APIFilePostOptions(file));
-	try {
-		await avatarFetch(url, options);
-	} catch (e) {
-		throw isAvatarAPIErrorResponse(e) ? toAvatarError(e) : e;
+	const url = '/aapi/avatar';
+	const payload = {
+		name: file.name,
+		type: file.type,
+		contents: (await base64FromFile(file)) as string,
+	};
+	// We send the request as a text/plain to avoid triggering the Express JSON parser
+	// which would try to parse the payload as JSON and fail. We instead parse the body
```

Comment:
is sending binary files as text a common / standard thing ?
is there no way to have an exception for a particular url and not try to parse it ?

Reply:
[There is](https://developer.mozilla.org/en-US/docs/Web/API/FormData), but we're doing this in this Base64 stringified format because the other file upload utility in MMA, the help centre contact us form, uses this format and the Express server is set up to handle upload bodies in that way. Rather than changing the server to handle two kinds of request body, we just make this one behave the same way as the other one.